loom/graphics/gfxTexture.cpp

```diff
@@ -294,24 +309,119 @@ TextureInfo *Texture::initFromBytes(utByteArray *bytes)
+    // Texture::enableAssetNotifications(true);
+
+    //Fire the async load complete delegate
+    tinfo->asyncLoadCompleteDelegate.invoke();
+    return 0;
+}
+
+
+TextureInfo * Texture::initFromAssetManagerAsync(const char *path)
 {
-        const int checkerboardSize = 128, checkSize = 8;        
+    TextureInfo *tinfo = NULL;
 
-        int *checkerboard = (int*)lmAlloc(gGFXTextureAllocator, checkerboardSize*checkerboardSize*4);
+    if (!path || !path[0])
+    {
+        lmLogError(gGFXTextureLogGroup, "Empty Path specified for initFromAssetManagerAsync()");
+        return NULL;
+    }
 
-        for (int i = 0; i < checkerboardSize; i++)
+    //check if texutre id already exists
+    //NOTE: shouldn't really happen... there is a check for this in LS that returns early there!    
+    TextureID *pid = Texture::sTexturePathLookup.get(path);
+    if(pid)
+    {
+        if ((*pid < 0) || (*pid >= MAXTEXTURES))
         {
-            for (int j = 0; j < checkerboardSize; j++)
-            {
-                checkerboard[(i * checkerboardSize) + j] = (((i / checkSize) ^ (j / checkSize)) & 1) == 0 ? 0xFF00FFFF : 0x00FF0077;
-            }
+            lmLogError(gGFXTextureLogGroup, "Invalid Texture ID returned in initFromAssetManagerAsync() for texture: %s", path);
+            return NULL;
+        }
+
+        loom_mutex_lock(Texture::sTexInfoLock);
+        tinfo = &Texture::sTextureInfos[*pid];
+        if (tinfo->handle.idx == bgfx::invalidHandle)
+        {
+            lmLogError(gGFXTextureLogGroup, "Invalid Texture Handle returned in initFromAssetManagerAsync() for texture: %s", path);
+            tinfo = NULL;
+        }
+        else
+        {
+            lmLog(gGFXTextureLogGroup, "Returning already reserved TextureInfo during initFromAssetManagerAsync() for texture: %s", path);
         }
+        loom_mutex_unlock(Texture::sTexInfoLock);
+        return tinfo;
+    }
+
+    // Get a new texture ID if one wasn't already found for the path
+    TextureID id = Texture::getAvailableTextureID();
+    if (id == TEXTUREINVALID)
+    {
+        lmLogWarn(gGFXTextureLogGroup, "No available texture id for %s", path);
+        return NULL;
+    }
+
+    // Initialize new texture info
+    loom_mutex_lock(Texture::sTexInfoLock);
+    tinfo = &Texture::sTextureInfos[id];
+    tinfo->handle.idx = MARKEDTEXTURE;    // mark in use, but not yet loaded
+    tinfo->texturePath = path;
+    Texture::sTexturePathLookup.insert(path, id);
+
+    //build up temp struct to use in the thread
+    AsyncLoadNote *threadNote = new AsyncLoadNote();
+    threadNote->id = id;
+    threadNote->path = path;
+    threadNote->tinfo = tinfo;
+    loom_mutex_unlock(Texture::sTexInfoLock);
+
+    //@Ben: This shouldn't be necessary for async image asset loading here like it is in scaleImageOnDisk right?
+    // Texture::enableAssetNotifications(false);
+
+    loom_thread_start(Texture::loadTextureAsync_body, (void *)threadNote);
```

Comment:
We definitely want to keep this from stacking (what if app has N images active at once)... so I think we will want to spawn a worker thread once and keep it around and fed via a queue.